gemini/sample-apps/accelerating_product_innovation/app/pages_utils/utils_product_features.py

```diff
@@ -0,0 +1,247 @@
+"""
+
+
+def _add_title_to_selection(title: str) -> None:
+    """Adds a title to the list of selected titles.
+
+    Args:
+        title (str): The title to add.
+    """
+    if title not in st.session_state.selected_titles:
+        st.session_state.selected_titles.append(title)
+        if st.session_state.modifying:
+            st.rerun()
+
+
+def _remove_title_from_selection(title: str) -> None:
+    """Removes a title from the list of selected titles.
+
+    Args:
+        title (str): The title to remove.
+    """
+    if title in st.session_state.selected_titles:
+        st.session_state.selected_titles.remove(title)
+
+
+def _render_box(box_id: str, title: str, parts: list, style: str) -> None:
+    """Renders a box with the given title, parts, and style.
+
+    Args:
+        box_id (str): The ID of the box.
+        title (str): The title of the box.
+        parts (list): The parts of the box.
+        style (str): The style of the box.
+    """
+    if len(parts) == 2:
+        st.markdown(
+            f'<div id="{box_id}" class="custom-box" style="{style}">'
+            f'<h5 style="color: #3367D6; text-align: center">{title}</h5>'
+            f"{parts[1]}</div>",
+            unsafe_allow_html=True,
+        )
+    else:
+        st.markdown(
+            f'<div id="{box_id}" class="custom-box" style="{style}">'
+            f"{parts[0]}</div>",
+            unsafe_allow_html=True,
+        )
+
+
+def get_points(text: str) -> list[str]:
+    """Gets a list of points from the given text.
+
+    Args:
+        text (str): The text to get the points from.
+
+    Returns:
+        list: A list of points.
+    """
+    points = text.split("\n")
+    curr = ""
+    sep_points = []
+    for point in points:
+        point = point.strip()
+        if point == "":
+            continue
+        if point.endswith(":"):
+            curr = point
+        else:
+            if point.endswith("."):
+                sep_points.append(curr.strip() + point.strip())
+                curr = ""
+            else:
+                curr += point
+    return sep_points
+
+
+def split_sentence(sentence: str) -> list[str]:
+    """Splits a sentence into two parts based on a colon (':') delimiter.
+
+    If the sentence doesn't contain a colon, the original sentence is returned
+    as a single item in a list.
+
+    Args:
+        sentence: The sentence to be split.
+
+    Returns:
+        A list of strings, containing either:
+            * The two parts of the sentence (if a colon was found)
+            * The original sentence as a single item (if no colon was found)
+    """
+
+    parts = sentence.split(":", 1)  # Split at most once
+
+    if len(parts) == 2:
+        # Extract and clean up the parts
+        first_part = parts[0].strip()
+        second_part = parts[1].strip()
+        return [first_part, second_part]
+
+    # No colon found, return the whole sentence
+    return [sentence]
+
+
+def trim_title(title: str) -> str:
```

Comment:
Seems like a very basic function. Is it really needed?

Reply:
Removed